### PR TITLE
Feat/cpp upload images on aws and add tags

### PR DESCRIPTION
**Pull request by kubovych**

### CODE REVIEW

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 1]
```diff
@@ -10,15 +13,28 @@ import classes from './styles.module.scss';
 interface Props {
   changePage: (currentPage: CurrentPage) => void
   currentPage: CurrentPage,
-  onChangeValue: (name: ProjectKeys, value: string) => void
+  onChangeValue: (name: ProjectKeys, value: any) => void
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: you can use type for it

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 2]
```diff
@@ -30,9 +46,30 @@ const Settings: FC<Props> = ({ changePage, currentPage, onChangeValue, region, i
...
+          label={t('Tags help when searching to give the user exactly those projects that interest him')}
+          onChange={e => setTag(e.target.value)}
+          value={tag}
+        />
+        <Button onClick={addTag} className={classes.addTag}>{t('Add')}</Button>
+      </div>
+      <div className={classes.listTags}>{tags.map(_tag => (
+        <Button
+          className={classes.tag}
+          key={_tag.tag.name}
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: can we have just tag.name?

File: backend/src/services/project/project.ts [hunk 3]
```diff
@@ -36,6 +46,10 @@ export default class ProjectService {
     const project:CreateProject = await this.#projectRepository.save({ ...body });
     const team = await this.#teamRepository.save({ ...new Team(), ...body.team, project });
     this.#chatRepository.save(body.team.chats.map(chat => ({ ...new Chat(), ...chat, team })));
+    const listTags = await this.#tagRepository.save(body.projectTags.map(
+      _tag => ({ ...new Tag(), name: _tag.tag.name })
+    ));
+    this.#projectTagRepository.save(listTags.map(_tag => ({ ...new ProjectTag(), tag: _tag, project })));
```
Comment (yevhenii-moroziuk) [inline backend/src/services/project/project.ts]: it would be nice to split logic between corresponding services

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

File: backend/src/services/project/project.ts [hunk 4]
```diff
@@ -45,6 +53,10 @@ export default class ProjectService {
     const project: CreateProject = await this.#projectRepository.save({ ...body });
     const team = await this.#teamRepository.save({ ...new Team(), ...body.team, project });
     this.#chatRepository.save(body.team.chats.map(chat => ({ ...new Chat(), ...chat, team })));
+    const listTags = await new TagService(this.#tagRepository).save(body.projectTags.map(projectTag => projectTag.tag));
```
Comment (yevhenii-moroziuk) [inline backend/src/services/project/project.ts]: Pass this services as dependencies

File: frontend/src/components/Input/index.tsx [hunk 5]
```diff
@@ -46,6 +49,18 @@ const Input = forwardRef<HTMLInputElement, Props>(({
               <FontAwesomeIcon icon={showPassword ? faEye : faEyeSlash} />
             </button>
           )}
+        {options && (
+        <div className={classes.options}>
+            {options.slice(0, 10).map(option => (
```
Comment (yevhenii-moroziuk) [inline frontend/src/components/Input/index.tsx]: What is 10? Let's use constant here

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 6]
```diff
@@ -10,15 +16,55 @@ import classes from './styles.module.scss';
...
+  onChangeValue: (name: ProjectKeys, value: string | CreateProjectTag[]) => void
   region: string,
-  imageUrl?:string,
+  imageUrl?: string,
+  videoUrl?: string,
+  newTags:CreateProjectTag[]
 }
 
-const Settings: FC<Props> = ({ changePage, currentPage, onChangeValue, region, imageUrl }) => {
+const Settings: FC<Props> = ({ changePage, currentPage, onChangeValue, region, imageUrl, videoUrl, newTags }) => {
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: Read this [article]( https://blog.bitsrc.io/best-practices-with-react-hooks-69d7e4af69a7#:~:text=3.%20Create%20your%20function%20components%20in%20the%20right%20order), please:

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 7]
```diff
@@ -10,15 +16,55 @@ import classes from './styles.module.scss';
...
   const { t } = useLocalization();
   const handleBack = () => changePage(currentPage - 1);
   const handleNext = () => changePage(CurrentPage.END);
+  const [tag, setTag] = useState('');
+  const addTag = () => {
+    if (newTags.indexOf({ tag: { name: tag } }) === -1) {
+      newTags.push({ tag: { name: tag } });
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: Do not mutate props. Create action/pass callback. Also, you can use newTags.find()

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 8]
```diff
@@ -30,9 +76,43 @@ const Settings: FC<Props> = ({ changePage, currentPage, onChangeValue, region, i
...
+        <Button onClick={addTag} className={classes.addTag}>{t('Add')}</Button>
+      </div>
+      <div className={classes.listTags}>{newTags.map(projectTag => (
+        <Button
+          className={classes.tag}
+          key={projectTag.tag.name}
+          id={projectTag.tag.name}
+          onClick={() => deleteTag(projectTag.tag.name)}
+        >
+          {`${projectTag.tag.name}`}
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: {projectTag.tag.name}

File: frontend/src/scenes/CreateProject/components/Settings/index.tsx [hunk 9]
```diff
@@ -30,9 +76,43 @@ const Settings: FC<Props> = ({ changePage, currentPage, onChangeValue, region, i
...
+        label={t('Atach video')}
+        onFileChange={file => onChangeValue(ProjectKeys.VIDEO_URL, file)}
+      />
+      <div className={classes.tagControl}>
+        <Input
+          type="text"
+          label={t('Tags help when searching to give the user exactly those projects that interest him')}
+          onChange={e => tagChanged(e.target.value)}
+          value={tag}
+          // onBlur={() => setOptions([])}
```
Comment (yevhenii-moroziuk) [inline frontend/src/scenes/CreateProject/components/Settings/index.tsx]: comment

**Review by yevhenii-moroziuk — CHANGES_REQUESTED** [review]

File: frontend/src/components/ImageUpload/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,8 @@
+@import '../../styles/vars';
+.addFile{
+    background: $color-root-background;
+    border: 1px solid $color-root-text;
+    &:hover{
+        cursor: pointer;
+    }
```
Comment (turfle) [inline frontend/src/components/ImageUpload/styles.module.scss]: Here it should be 2 spaces offset, not 4

File: frontend/src/components/Input/index.tsx [hunk 11]
```diff
@@ -46,6 +51,18 @@ const Input = forwardRef<HTMLInputElement, Props>(({
               <FontAwesomeIcon icon={showPassword ? faEye : faEyeSlash} />
             </button>
           )}
+        {options && (
+        <div className={classes.options}>
+            {options.slice(0, countOptions).map(option => (
```
Comment (turfle) [inline frontend/src/components/Input/index.tsx]: 2 spaces

**Review by turfle — APPROVED** [review]

**Review by yevhenii-moroziuk — APPROVED** [review]

**Pull request merged**